archaius-core/src/main/java/com/netflix/config/PropertyWrapper.java

```diff
@@ -158,9 +159,11 @@ public void addCallback(Runnable callback) {
      */
     @Override
     public void removeAllCallbacks() {
-        for (Runnable callback: callbackList) {
+        final Set<Runnable> callbacksToRemove = new HashSet<Runnable>(callbacks);
+        for (Runnable callback : callbacksToRemove) {
             prop.removeCallback(callback);
         }
+        callbacks.removeAll(callbacksToRemove);
```

Comment:
Since other threads could add new callbacks while `removeAllCallbacks` is processing, this is just extra protection to prevent removing callbacks from the set that haven't had `removeCallback(Runnable)` called on them.